school/models/school.py

```diff
@@ -525,20 +583,40 @@ class SubjectSubject(models.Model):
     is_practical = fields.Boolean(
         "Is Practical", help="Check this if subject is practical."
     )
-    elective_id = fields.Many2one(
-        "subject.elective",
-        help="""Elective subject respective
-                                  the following subject""",
+    # student_ids = fields.Many2many('student.student',
+    #                                'elective_subject_student_rel',
+    #                                'subject_id', 'student_id', 'Students',
+    #                                help='Students who choose this subject')
+    is_elective_subject = fields.Boolean(
+        "Is Elective Subject", help="Check this if subject is elective."
     )
-    student_ids = fields.Many2many(
+    student_ids = fields.One2many(
         "student.student",
-        "elective_subject_student_rel",
         "subject_id",
-        "student_id",
-        "Students",
-        help="Students who choose this subject",
+        "Student In Class",
+        compute="_compute_student_subject",
+        help="Students which are in this standard",
     )
 
+    @api.model_create_multi
+    def create(self, values):
+        result = super(SubjectSubject, self).create(values)
+        for standard in self.env["school.standard"].search([]):
+            standard.onchange_subject_related_standard()
+        return result
+
+    def write(self, values):
+        res = super(SubjectSubject, self).write(values)
+        if values.get("standard_id"):
+            for standard in self.env["school.standard"].search([]):
+                standard.onchange_subject_related_standard()
+        return res
```

Comment:
why calling onchange on create n write ? should nt it be enough to use under api.onchange standrd id?

Reply:
@JayVora-SerpentCS  Sir if subject record have change or add the standard it will automatically reflect in school standard records at create and write method. user can not remove or add subject manually in school standard record every time.